corehq/apps/app_manager/templates/app_manager/form_view_base.html

```diff
@@ -103,9 +103,18 @@
             var isModuleFilterEnabled = {{ is_module_filter_enabled|BOOL }};
             var formFilter = {{ form.form_filter|JSON }};
             var allOtherFormsRequireACase = {{ form.all_other_forms_require_a_case|BOOL }};
+            var formFilterReferencesCase = {{ form_filter_references_case|BOOL }};
             var formFilterAllowed = ko.computed(function () {
```

Comment:
great, thanks @kaapstorm.  so then the proposed behavior becomes:

- always allow entering form filter conditions
- raise an error during build if:
    - a form filter condition is entered that references a case but the module is set to "show only forms" or not all of the forms in the module use a case
    - a form filter condition is entered that references a user case but the domain doesn't use user as a case

(an error will also be displayed in the form filter ui once the page is refreshed, doing so without a refresh involves a little more work)

@dimagi/product  can you confirm that the logic outlined above is how we want form filter validation to behave?